lib/pbench/server/api/resources/datasets_contents.py

```diff
@@ -0,0 +1,160 @@
+        if details.type is CacheType.DIRECTORY:
+            children = info["children"] if "children" in info else {}
+            dir_list = []
+            file_list = []
+
+            for c, value in children.items():
+                d: CacheObject = value["details"]
+                if d.type is CacheType.DIRECTORY:
+                    dir_list.append(
+                        {
+                            "name": c,
+                            "type": d.type.name,
+                            "uri": f"{origin}/contents/{d.location}",
+                        }
+                    )
+                elif d.type in (CacheType.FILE, CacheType.OTHER):
+                    file_list.append(
+                        {
+                            "name": c,
+                            "size": d.size,
+                            "type": d.type.name,
+                            "uri": f"{origin}/inventory/{d.location}",
+                        }
+                    )
+                elif d.type is CacheType.SYMLINK:
+                    if d.resolve_type is CacheType.DIRECTORY:
+                        access = "contents"
+                    else:
+                        access = "inventory"
+                    file_list.append(
+                        {
+                            "name": c,
+                            "type": d.type.name,
+                            "link": str(d.resolve_path),
+                            "link_type": d.resolve_type.name,
+                            "uri": f"{origin}/{access}/{d.resolve_path}",
+                        }
+                    )
+
+            dir_list.sort(key=lambda d: d["name"])
+            file_list.sort(key=lambda d: d["name"])
+
+            # Normalize because we want the "root" directory to be reported as
+            # "" rather than as Path's favored "."
+            loc = "" if str(details.location) == "." else str(details.location)
+            val = {
+                "name": details.name,
+                "type": details.type.name,
+                "directories": dir_list,
+                "files": file_list,
+                "uri": f"{origin}/contents/{loc}",
+            }
+        else:
+            access = "inventory"
+            if details.type is CacheType.SYMLINK:
+                link = str(details.resolve_path)
+                if details.resolve_type is CacheType.DIRECTORY:
+                    access = "contents"
+            else:
+                link = str(details.location)
+            val = {
+                "name": details.name,
+                "type": details.type.name,
+                "uri": f"{origin}/{access}/{link}",
+            }
+            if details.type is CacheType.SYMLINK:
+                val["link"] = str(details.resolve_path)
+                val["link_type"] = details.resolve_type.name
+            elif details.type is CacheType.FILE:
+                val["size"] = details.size
```

Comment:
Yeah, I remember throwing the `CacheType.OTHER` into the tuple above and deciding not to worry about it rather than make a special case. But I don't think "size" is a meaningful concept here aside from `FILE`, so maybe I need to break up the `dict` up above.